.travis.yml

```diff
@@ -4,38 +4,128 @@ notifications:
   email: false
 language: node_js
 node_js:
-  - "10"
   - "8"
-  - "6"
-jdk:
-  - oraclejdk9
-  - openjdk8
-env:
-  - MAVEN_VERSION=3.6.1 JAVA_VERSION=openjdk8
-  - MAVEN_VERSION=3.6.1 JAVA_VERSION=oraclejdk9
 
-  - MAVEN_VERSION=3.3.9 JAVA_VERSION=oraclejdk9
-  - MAVEN_VERSION=3.3.9 JAVA_VERSION=openjdk8
+cache:
+  directories:
+    - node_modules
+    - $HOME/.m2
 
 before_install:
-  # sdkman helps to install different gradle versions
+  # sdkman helps to install different maven versions
   - curl -s "https://get.sdkman.io" | bash
   - source "$HOME/.sdkman/bin/sdkman-init.sh"
   - echo sdkman_auto_answer=true > ~/.sdkman/etc/config
-cache:
-  directories:
-    - node_modules
-script:
-  - jdk_switcher use $JAVA_VERSION
-  - sdk install maven $MAVEN_VERSION
-  - npm test
+  # install jdk is needed for JDK10+
+  - wget https://github.com/sormuras/bach/raw/master/install-jdk.sh
+
 jobs:
   include:
+    # test MAVEN_VERSION=3.6.1
+    - stage: Node versions + actual Maven version
```

Comment:
for this stage can we just have:
node 6 => java 9
nod 8 =>  java 10
node 10 =>  java 11

Just to test js code, and then the next stage is all the maven versions. Just to reduce the number of tests we run

Reply:
```
- stage: Node versions + actual Maven version
      node_js: 6
      env: MAVEN_VERSION=3.6.1 JDK=9
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      node_js: 8
      env: MAVEN_VERSION=3.6.1 JDK=10
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      node_js: 10
      env: MAVEN_VERSION=3.6.1 JDK=11
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    - stage: Older Maven version
      env: MAVEN_VERSION=3.3.9 JDK=8
      script:
        - jdk_switcher use openjdk8
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      env: MAVEN_VERSION=3.3.9 JDK=9
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      env: MAVEN_VERSION=3.3.9 JDK=10
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      env: MAVEN_VERSION=3.3.9 JDK=11
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      env: MAVEN_VERSION=3.3.9 JDK=12
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
    -
      env: MAVEN_VERSION=3.3.9 JDK=13
      script:
        - source install-jdk.sh --feature $JDK
        - ln -sf /etc/ssl/certs/java/cacerts $JAVA_HOME/lib/security/cacerts
        - sdk install maven $MAVEN_VERSION
        - java -version
        - npm test
```

Does this sound better?
### stage 1 (current maven version - 3.6.1):
- node 6 => java 9
- node 8 => java 10
- node 10 => java 11
### stage 2 (3.3.9 + 3.6.1)
- node 8 => java 8 (3.3.9)
- node 8 => java 10 (3.3.9)
- node 8 => java 11 (3.3.9)
- node 8 => java 12 (3.3.9)
- node 8 => java 13 (3.3.9)
- node 8 => java 8 (3.6.1)

Let me know if we want it differently? 

P.S.: I've duplicated by accident stage: Node versions + actual Maven version.
I've corrected in the above snippet.

Reply:
maybe `node 8 => java 12 (3.6.1)`? looks great otherwise

Reply:
Done.

Reply:
😍